app/src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -381,7 +415,21 @@ public boolean onOptionsItemSelected(MenuItem item) {
                             .show();
                     return true;
                 }
-                getApi().addFolder(mFolder);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP &&
+                    mFolderUri != null) {
```

Comment:
Should we only do this for send only folders?

Reply:
Yes, we pre-check if syncthing can write to a folder and only then, the user can move the slider to enable sendreceive mode. If syncthing can write itself, it will create .stfolder correctly. So it is sufficient to do the creation of .stfolder from the java code only for send only folders as you suggested. I'll commit that shortly.